app/rtc_compare.py

```diff
@@ -0,0 +1,396 @@
+'''
+Compare two RTC products if they are equivalent
+Part of the codes are copied from PROTEUS SAS
+'''
+
+import argparse
+import itertools
+
+import h5py
+import numpy as np
+
+RTC_S1_PRODUCTS_ERROR_TOLERANCE = 1e-6
+
+def _get_parser():
+    parser = argparse.ArgumentParser(
+        description='Compare two RTC products',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    # Inputs
+    parser.add_argument('input_file',
+                        type=str,
+                        nargs=2,
+                        help='Input RTC products in NETCDF/HDF5 format')
+
+    return parser
+
+def print_data_difference(val_1, val_2, indent=4):
+    '''
+    Print out the diffrence of the data whose dimension is >= 1
+
+    Parameters:
+    -----------
+    val_1, val_2: np.array
+        Data that has difference to each other
+    indent: int
+        Number of spaces for indentation
+    '''
+
+    str_indent = ' ' * indent + '-'
+
+    # printout the difference when the array is numeric
+    if issubclass(val_1.dtype.type, np.number) and issubclass(val_2.dtype.type, np.number):
+        difference_val = val_1 - val_2
+        index_max_diff = np.nanargmax(np.abs(difference_val))
+
+        print(f'{str_indent} Maximum difference detected from index '
+              f'[{index_max_diff}]: '
+              f'1st: ({val_1[index_max_diff]}), 2nd: ({val_2[index_max_diff]}) = '
+              f'diff: ({difference_val[index_max_diff]})')
+    else:
+        # Non-numerical array
+        flag_discrepancy = (val_1 != val_2)
+        index_first_discrepancy = np.where(flag_discrepancy)[0][0]
+        print(f'{str_indent} The first discrepancy has detected from index '
+              f'[{index_first_discrepancy}] : '
+              f'1st=({val_1[index_first_discrepancy]}), 2nd=({val_2[index_first_discrepancy]})')
+
+    # Check pixel-by-pixel nan / non-nan difference
+    if (issubclass(val_1.dtype.type, np.floating) and issubclass(val_2.dtype.type, np.floating)) or\
+    (issubclass(val_1.dtype.type, np.complexfloating) and issubclass(val_2.dtype.type, np.complexfloating)):
+
+        mask_nan_val_1 = np.isnan(val_1)
+        mask_nan_val_2 = np.isnan(val_2)
+
+        mask_nan_discrepancy = np.logical_xor(mask_nan_val_1, mask_nan_val_2)
+
+        if np.any(mask_nan_discrepancy):
```

Comment:
Please, use early escaping here. If there's no discrepancy, you can just exit

Reply:
How about this?

```python
if not np.any(mask_nan_discrepancy):
    print('NaN discrepancy was not detected.')

num_pixel_nan_discrepancy = mask_nan_discrepancy.sum()
index_pixel_nan_discrepancy = np.where(mask_nan_discrepancy)
print(f'{str_indent} Found {num_pixel_nan_discrepancy} of '
'NaN inconsistecy between the input arrays. '
f'First index of the discrepancy: [{index_pixel_nan_discrepancy[0][0]}]')
...
```

Reply:
Yes, and you need a return after closing the `if` statement above.